src/Reader.php

```diff
@@ -52,6 +52,20 @@ public function query(callable $callable = null)
         $iterator = $this->applyIteratorFilter($iterator);
         $iterator = $this->applyIteratorSortBy($iterator);
         $iterator = $this->applyIteratorInterval($iterator);
+
```

Comment:
By moving it Up you ensure that the BOM sequence will be indeed remove from the First line of the document. After the filters have been applied you may be remove a valid BOM sequence inside the CSV